tools/functions.sh

```diff
@@ -1,14 +1,15 @@
+#!/bin/sh
 ###############################################################
 # SPDX-License-Identifier: BSD-2-Clause-Patent
 # Copyright (c) 2019 Tomer Eliyahu (Intel)
 # This code is subject to the terms of the BSD+Patent license.
 # See LICENSE file for more details.
 ###############################################################
 
-installdir="${rootdir}/build/install"
+export installdir="${rootdir:?}/build/install"
```

Comment:
We really don't want to export it. This is an artifact of it being sourced. So here we need an exception.

rootdir is expected to have been set by the calling script. Instead of giving it a default, add a condition that errors out if it is not set.

Reply:
1) about `installdir` - agree. I'll add exception
2) about `rootdir`
`${rootdir:?}` does not define default value for variable but check that variable was set

Usage example
Add file `test.sh`
```
#!/bin/sh
. tools/functions.sh
echo "Hi"
```
and try execute it
```
$ sh test.sh 
test.sh: 9: tools/functions.sh: rootdir: parameter not set or null
```

Reply:
Oh, I didn't know that. Excellent!